lib/Dialect/SPIRV/Serialization/Deserializer.cpp

```diff
@@ -722,6 +749,46 @@ LogicalResult Deserializer::processFunctionType(ArrayRef<uint32_t> operands) {
   return success();
 }
 
+LogicalResult Deserializer::processStructType(ArrayRef<uint32_t> operands) {
+  if (operands.size() < 2) {
+    return emitError(unknownLoc, "OpTypeStruct must have at least one member");
+  }
+
+  SmallVector<Type, 0> memberTypes;
+  for (uint32_t opNum = 1; opNum < operands.size(); ++opNum) {
+    Type memberType = getType(operands[opNum]);
+    if (!memberType) {
+      return emitError(unknownLoc, "OpTypeStruct references undefined <id> ")
+             << operands[opNum];
+    }
+    memberTypes.push_back(memberType);
+  }
+
+  // Check for layoutinfo
+  auto memberDecorationIterator = memberDecorationMap.find(operands[0]);
+  if (memberDecorationIterator != memberDecorationMap.end()) {
+    // Each member must have an offset
+    SmallVector<spirv::StructType::LayoutInfo, 0> layoutInfo;
+    const auto &offsetDecorationMap = memberDecorationIterator->second;
+    for (uint32_t memberNum = 0; memberNum < memberTypes.size(); ++memberNum) {
```

Comment:
Please follow LLVM style to avoid recomputing the end of a loop,
https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop

Reply:
My fault, thanks for mention this!